### PR TITLE
Integrate development guide with lockpicking lab setup

### PR DESCRIPTION
**Pull request by nd-0r**
Since we already link to the development guide in the Lockpicking lab docs, this PR moves the VM setup guide to the development guide and integrates the changes.

- Adds PAT authentication docs to development guide
- Adds VM connection docs to development guide
- Adds Git configuration docs to development guide
- Removes VM/Git configuration docs from Lockpicking lab docs

### CODE REVIEW

File: _tutorials/development.md [hunk 1]
```diff
@@ -22,32 +25,67 @@ $ ssh <NETID>@{{site.data.constants.semester }}-{{ site.data.constants.departmen
...
+remote: Support for password authentication was removed on August 13, 2021.
+remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
+fatal: Authentication failed for 'https://github.com/illinois-cs-coursework/{{site.data.constants.semester }}_{{ site.data.constants.department_code }}{{site.data.constants.course_number}}_NETID.git'
+```
+
+### Authenticating with GitHub
+
+There are three ways that we recommend for authenticating with GitHub
+
+#### Option 1: Using SSH keys
```
Comment (nsunkad) [inline _tutorials/development.md:76]: Maybe hyperlink "SSH keys" to the official Github article on how to use ssh keys 
[](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)

File: _tutorials/development.md [hunk 2]
```diff
@@ -56,8 +94,20 @@ In summary:
 1. Install the `gh` tool using `sudo apt update && sudo apt install gh`.
 2. Run `gh auth login` to authenticate.
 
+#### Option 3: Using a Personal Access Token
```
Comment (nsunkad) [inline _tutorials/development.md:97]: Perhaps hyperlink to https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/authorizing-a-personal-access-token-for-use-with-saml-single-sign-on

File: _tutorials/development.md (hunk 1 above)
Comment (nd-0r) [reply _tutorials/development.md]: Updated

File: _tutorials/development.md [hunk 3]
```diff
@@ -22,32 +25,67 @@ $ ssh <NETID>@{{site.data.constants.semester }}-{{ site.data.constants.departmen
...
+
+There are three ways that we recommend for authenticating with GitHub
+
+#### Option 1: Using SSH keys
 
 GitHub provides an easy way to add SSH keys to your account. GitHub's documentation can be found [here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account?platform=linux). This method has the added benefit that you can reuse the same SSH key for password-less authentication to your CS341 VM.
 
 In summary:
 
-1. Create an SSH key on your local machine if you do not already have one. Currently, we recommend the EdDSA key algorithm. You can create an EdDSA key with the command `ssh-keygen -t ed25519 -a 100 -f`.
```
Comment (nsunkad) [inline _tutorials/development.md:44]: "This command generates a private and public key for authentication. It will prompt you to enter a name for the file  storing your private key, and will generate a .pub file for your public key." 

maybe?

Comment (nd-0r) [reply _tutorials/development.md:44]: That's a good idea, it isn't clear exactly what the command does. Updated.

File: _tutorials/development.md (hunk 2 above)
Comment (nd-0r) [reply _tutorials/development.md:97]: Updated

**Review by xwei16 — APPROVED** [review]
Lgtm!
